src/Tool/Telemetry.cs

```diff
@@ -228,6 +235,11 @@ public void SetSharedContextIfChanged(IMetadataController metadataController, st
 
     public static class TelemetryExtensions
     {
+        // Generate a new UUID to use as a session ID.
+        // By making a static property of a non-generic static class, we
+        // ensure that there's one session ID per process.
+        internal static readonly Guid SessionId = Guid.NewGuid();
```

Comment:
I wonder if we want to rename this to KernelInstanceId or something like that? I am just finishing up a privacy review, and they typically categorize SessionIds as EUPI because it's usually referring to user's login session. Since we are not persisting it according to login sessions, it may be nice to avoid confusion or raised flags around storing SessionIds, and go with term that'll cause less alarm for privacy reviewers.

Reply:
Good idea, let me go on and rename that, then.